### PR TITLE
Improve InfluxdbWriter performance

### PR DESCRIPTION
**Pull request by gunnarbeutner**
This also depends on PR #5759 and #5760.

### CODE REVIEW

File: lib/perfdata/influxdbwriter.cpp [hunk 1]
```diff
@@ -381,45 +358,60 @@ void InfluxdbWriter::SendMetric(const Dictionary::Ptr& tmpl, const String& label
...
 				msgbuf << ",";
 
-			msgbuf << EscapeKey(pair.first) << "=" << EscapeField(pair.second);
+			msgbuf << EscapeKeyOrTagValue(pair.first) << "=" << EscapeValue(pair.second);
 		}
 	}
 
 	msgbuf << " " <<  static_cast<unsigned long>(ts);
 
+#ifdef _DEBUG
```
Comment (dnsmichi) [inline lib/perfdata/influxdbwriter.cpp]: Should be `I2_DEBUG`.

File: lib/perfdata/influxdbwriter.cpp [hunk 2]
```diff
@@ -381,45 +358,60 @@ void InfluxdbWriter::SendMetric(const Dictionary::Ptr& tmpl, const String& label
...
+		double st = Utility::GetTime();
+
+		try {
+			Flush();
+		} catch (...) {
+			/* Do nothing. */
+		}
+
+		double et = Utility::GetTime();
+		Log(LogInformation, "InfluxDbWriter")
```
Comment (dnsmichi) [inline lib/perfdata/influxdbwriter.cpp]: That might be too noisy as information level.

Comment (gunnarbeutner) [reply lib/perfdata/influxdbwriter.cpp]: Oops, that's not even meant to be there at all, at least not in the PR. 😇

**Comment by dnsmichi** [comment]
That's de facto a rewrite of the InfluxDB feature. Must be tested with care before any release.

**Comment by gunnarbeutner** [comment]
Aye, this is definitely a major change.

**Comment by dnsmichi** [comment]
Backporting will be "funny" with the code quality changes applied in master.

**Comment by dnsmichi** [comment]
5759 and #5760 have been merged to master, a test backport to 2.8 went fine. Now for rebasing this PR.

**Comment by gunnarbeutner** [comment]
Backporting as per @lippserd's request. :)

**Comment by dnsmichi** [comment]
We agreed on reverting this from 2.8.x as the InfluxDBWriter performance fixes require more reliable and long-term tests. Such changes will only hit master and 2.9 then.

Required reverts:

#5759 
#5760 
#5764 **[x]**

**Pull request merged**